test/e2e/features/install.feature

```diff
@@ -0,0 +1,272 @@
+    Given "extra" catalog serves bundles
+    When ClusterExtension is applied
+      """
+      apiVersion: olm.operatorframework.io/v1
+      kind: ClusterExtension
+      metadata:
+        name: $NAME
+      spec:
+        namespace: $TEST_NAMESPACE
+        serviceAccount:
+          name: olm-sa
+        source:
+          sourceType: Catalog
+          catalog:
+            packageName: test
+      """
+    Then ClusterExtension reports Progressing as True with Reason Retrying:
+      """
+      found bundles for package "test" in multiple catalogs with the same priority [extra-catalog test-catalog]
+      """
+
+  @SingleOwnNamespaceInstallSupport
+  Scenario: watchNamespace config is required for extension supporting single namespace
+    Given service account "olm-admin" in test namespace is cluster admin
+    And resource is applied
+      """
+      apiVersion: v1
+      kind: Namespace
+      metadata:
+        name: single-namespace-operator-target
+      """
+    And ClusterExtension is applied
+      """
+      apiVersion: olm.operatorframework.io/v1
+      kind: ClusterExtension
+      metadata:
+        name: $NAME
+      spec:
+        namespace: $TEST_NAMESPACE
+        serviceAccount:
+          name: olm-admin
+        source:
+          sourceType: Catalog
+          catalog:
+            packageName: single-namespace-operator
+            selector:
+              matchLabels:
+                "olm.operatorframework.io/metadata.name": test-catalog
+      """
+    And ClusterExtension reports Progressing as True with Reason Retrying:
+      """
+      error for resolved bundle "single-namespace-operator.1.0.0" with version "1.0.0":
+      invalid ClusterExtension configuration: invalid configuration: required field "watchNamespace" is missing
+      """
+    When ClusterExtension is updated
+      """
+      apiVersion: olm.operatorframework.io/v1
+      kind: ClusterExtension
+      metadata:
+        name: $NAME
+      spec:
+        namespace: $TEST_NAMESPACE
+        serviceAccount:
+          name: olm-admin
+        config:
+          configType: Inline
+          inline:
+            watchNamespace: single-namespace-operator-target # added
+        source:
+          sourceType: Catalog
+          catalog:
+            packageName: single-namespace-operator
+            selector:
+              matchLabels:
+                "olm.operatorframework.io/metadata.name": test-catalog
+      """
+    Then ClusterExtension reports Installed as True
+    And bundle "single-namespace-operator.1.0.0" is installed in version "1.0.0"
+    And operator "single-namespace-operator" target namespace is "single-namespace-operator-target"
+
+  @SingleOwnNamespaceInstallSupport
+  Scenario: watchNamespace config is required for extension supporting own namespace
+    Given service account "olm-admin" in test namespace is cluster admin
+    And ClusterExtension is applied
```

Comment:
nit: would be cool if we could call out that the ClusterExtension is applied without the watchNamespace configuration, or something like that